src/app/components/hooks/use-model-state.ts

```diff
@@ -166,27 +219,45 @@ export const useModelState = <IModelInputState, IModelOutputState>(
     setActiveRunIdx(idx);
     const run = modelRuns[idx];
     const snapshots = run.outputStateSnapshots;
-    setCurrentOutputState(snapshots[snapshots.length - 1]);
+
+    setCurrentOutputState(() => {
+      saveInteractiveState({outputState: snapshots[snapshots.length - 1]});
```

Comment:
Same question here about zero length of the snapshots array?  Maybe that is never the case?

Reply:
The snapshots array always has at least a length of 1. It is always initialized with a the defaultOutputState.